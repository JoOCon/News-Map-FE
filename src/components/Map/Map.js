import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactMapboxGl from 'react-mapbox-gl';
import Events from '../Events/Events';

import { getEvents } from '../../utilities/apiCalls/apiCalls';
import { mbAccessToken as TOKEN } from '../../utilities/apiCalls/apiKeys';

export class Map extends Component {
  constructor() {
    super();
    this.state = {
      latitude: 0,
      longitude: 0,
      events: []
    };
  }

  componentDidMount() {
    this.setLatLngEvents();
  }

  setLatLngEvents = async () => {
    await navigator.geolocation.getCurrentPosition(async location => {
      const { latitude, longitude } = location.coords;
      this.retrieveEvents(latitude, longitude);
      this.setState({
        latitude,
        longitude
      });
    });
  };

  retrieveEvents = async (lat, lng) => {
    const events = await getEvents(lat, lng);
    await this.setState({ events });
  };

  render() {
    const Map = ReactMapboxGl({
      accessToken: TOKEN
    });
    const { latitude, longitude, events } = this.state;
    return (
      <Map
        center={[longitude, latitude]}
        zoom={[13]}
        style={`mapbox://styles/mapbox/${this.props.mapStyle}-v9`}
        containerStyle={{
          height: '100vh',
          width: '100vw'
        }}
      >
        <Events events={events} />
      </Map>
    );
  }
}

Map.propTypes = {
  latitude: PropTypes.number,
  longitude: PropTypes.number,
  events: PropTypes.array
};

export const mapDispatchToProps = () => ({
  setEvents: events => dispatch(setEvents(events))
});

export default connect(
  null,
  mapDispatchToProps
)(Map);
