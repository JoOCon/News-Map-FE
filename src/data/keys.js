export const googleClientId =
  '241779944002-qa42v6cvk3kpu8da0jimdr364vn41t1r.apps.googleusercontent.com';
