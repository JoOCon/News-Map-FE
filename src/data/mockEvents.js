export const eventsResponse = [
  {
    name: 'P!NK: BEAUTIFUL TRAUMA WORLD TOUR',
    e_id: 'G5vzZ4amDV3z0',
    url: 'https://www.altitudetickets.com/events/detail/pink-2019',
    img:
      'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg',
    date: 'Mon, Apr 1, 2019 7:30 PM',
    venue_name: 'Pepsi Center',
    address: '1000 Chopper Circle Denver, CO 80204',
    lat: '39.74724',
    lng: '-105.010166',
    distance: 0.75,
    favorite: false
  },
  {
    name: 'Childish Gambino',
    e_id: 'G5vzZ4aV5O3CO',
    url: 'https://www.altitudetickets.com/events/detail/childish-gambino',
    img:
      'https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg',
    date: 'Tue, Dec 4, 2018 7:30 PM',
    venue_name: 'Pepsi Center',
    address: '1000 Chopper Circle Denver, CO 80204',
    lat: '39.74724',
    lng: '-105.010166',
    distance: 0.75,
    favorite: true
  }
];

export const eventStoreResponse = [
  {
    name: 'P!NK: BEAUTIFUL TRAUMA WORLD TOUR',
    e_id: 'G5vzZ4amDV3z0',
    url: 'https://www.altitudetickets.com/events/detail/pink-2019',
    img:
      'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg',
    date: 'Mon, Apr 1, 2019 7:30 PM',
    venue_name: 'Pepsi Center',
    address: '1000 Chopper Circle Denver, CO 80204',
    lat: '39.74724',
    lng: '-105.010166',
    distance: 0.75,
    favorite: false
  },
  {
    name: 'Childish Gambino',
    e_id: 'G5vzZ4aV5O3CO',
    url: 'https://www.altitudetickets.com/events/detail/childish-gambino',
    img:
      'https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg',
    date: 'Tue, Dec 4, 2018 7:30 PM',
    venue_name: 'Pepsi Center',
    address: '1000 Chopper Circle Denver, CO 80204',
    lat: '39.74724',
    lng: '-105.010166',
    distance: 0.75,
    favorite: false
  }
];

export const favoriteEventResponse = {
  message:
    'Successfully added EnParejaDos - con Adrian Uribe y Consuelo Duval with id: G5vzZ48H7Q7cH'
};

export const mockTargetEventFalse = {
  name: 'Danzig - 30 Year Anniversary Tour',
  e_id: 'G5vzZ4C58cdb-',
  url: 'http://concerts.livenation.com/event/1E0054E2D0B43621',
  img:
    'https://s1.ticketm.net/dam/a/492/732ca6af-f94f-477d-9e66-5f56f9e54492_450551_EVENT_DETAIL_PAGE_16_9.jpg',
  date: 'Wed, Oct 31, 2018 6:30 PM',
  venue_name: 'Fillmore Auditorium (Denver)',
  address: '1510 Clarkson Denver, CO 80218',
  lat: '39.740611',
  lng: '-104.977194',
  distance: 0.5,
  favorite: false
};

export const mockTargetEventTrue = {
  name: 'Danzig - 30 Year Anniversary Tour',
  e_id: 'G5vzZ4C58cdb-',
  url: 'http://concerts.livenation.com/event/1E0054E2D0B43621',
  img:
    'https://s1.ticketm.net/dam/a/492/732ca6af-f94f-477d-9e66-5f56f9e54492_450551_EVENT_DETAIL_PAGE_16_9.jpg',
  date: 'Wed, Oct 31, 2018 6:30 PM',
  venue_name: 'Fillmore Auditorium (Denver)',
  address: '1510 Clarkson Denver, CO 80218',
  lat: '39.740611',
  lng: '-104.977194',
  distance: 0.5,
  favorite: true
};

export const mockTargetEventFalseTwo = {
  name:
    'Danzig - 30 Year Anniversary Tourasdsadasdsadasdasdasdsadsadasdsadasdasdasd',
  e_id: 'G5vzZ4C58cdb-',
  url: 'http://concerts.livenation.com/event/1E0054E2D0B43621',
  img:
    'https://s1.ticketm.net/dam/a/492/732ca6af-f94f-477d-9e66-5f56f9e54492_450551_EVENT_DETAIL_PAGE_16_9.jpg',
  date: 'Wed, Oct 31, 2018 6:30 PM',
  venue_name: 'Fillmore Auditorium (Denver)',
  address: '1510 Clarkson Denver, CO 80218',
  lat: '39.740611',
  lng: '-104.977194',
  distance: 0.5,
  favorite: true
};

export const mockTargetEventTrueTwo = [
  {
    name: 'Danzig - 30 Year Anniversary Toqdqffsdfasdfasfasdur',
    e_id: 'G5vzZ4C58cdb-',
    url: 'http://concerts.livenation.com/event/1E0054E2D0B43621',
    img:
      'https://s1.ticketm.net/dam/a/492/732ca6af-f94f-477d-9e66-5f56f9e54492_450551_EVENT_DETAIL_PAGE_16_9.jpg',
    date: 'Wed, Oct 31, 2018 6:30 PM',
    venue_name: 'Fillmore Auditorium (Denver)',
    address: '1510 Clarkson Denver, CO 80218',
    lat: '39.740611',
    lng: '-104.977194',
    distance: 0.5,
    favorite: true
  }
];

export const mockEventsResponse = {
  _embedded: {
    events: [
      {
        name: 'P!NK: BEAUTIFUL TRAUMA WORLD TOUR',
        type: 'event',
        id: 'G5vzZ4amDV3z0',
        test: false,
        url: 'https://www.altitudetickets.com/events/detail/pink-2019',
        locale: 'en-us',
        images: [
          {
            ratio: '3_2',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg',
            width: 640,
            height: 427,
            fallback: false
          },
          {
            ratio: '4_3',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_CUSTOM.jpg',
            width: 305,
            height: 225,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_16_9.jpg',
            width: 640,
            height: 360,
            fallback: false
          },
          {
            ratio: '3_2',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_ARTIST_PAGE_3_2.jpg',
            width: 305,
            height: 203,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_LANDSCAPE_16_9.jpg',
            width: 1136,
            height: 639,
            fallback: false
          },
          {
            ratio: '3_2',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_3_2.jpg',
            width: 1024,
            height: 683,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RECOMENDATION_16_9.jpg',
            width: 100,
            height: 56,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_LARGE_16_9.jpg',
            width: 2048,
            height: 1152,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_EVENT_DETAIL_PAGE_16_9.jpg',
            width: 205,
            height: 115,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_16_9.jpg',
            width: 1024,
            height: 576,
            fallback: false
          }
        ],
        distance: 1.57,
        units: 'MILES',
        sales: {
          public: {
            startDateTime: '2018-05-11T16:00:00Z',
            startTBD: false,
            endDateTime: '2019-04-02T01:30:00Z'
          }
        },
        dates: {
          start: {
            localDate: '2019-04-01',
            localTime: '19:30:00',
            dateTime: '2019-04-02T01:30:00Z',
            dateTBD: false,
            dateTBA: false,
            timeTBA: false,
            noSpecificTime: false
          },
          timezone: 'America/Denver',
          status: { code: 'onsale' },
          spanMultipleDays: false
        },
        classifications: [
          {
            primary: true,
            segment: { id: 'KZFzniwnSyZfZ7v7nJ', name: 'Music' },
            genre: { id: 'KnvZfZ7vAeA', name: 'Rock' },
            subGenre: { id: 'KZazBEonSMnZfZ7v6dt', name: 'Alternative Rock' },
            type: { id: 'KZAyXgnZfZ7v7nI', name: 'Undefined' },
            subType: { id: 'KZFzBErXgnZfZ7v7lJ', name: 'Undefined' },
            family: false
          }
        ],
        promoter: {
          id: '653',
          name: 'LIVE NATION MUSIC',
          description: 'LIVE NATION MUSIC / NTL / USA'
        },
        promoters: [
          {
            id: '653',
            name: 'LIVE NATION MUSIC',
            description: 'LIVE NATION MUSIC / NTL / USA'
          }
        ],
        _links: {
          self: { href: '/discovery/v2/events/G5vzZ4amDV3z0?locale=en-us' },
          attractions: [
            { href: '/discovery/v2/attractions/K8vZ9171Jo7?locale=en-us' }
          ],
          venues: [{ href: '/discovery/v2/venues/KovZpZAFaJeA?locale=en-us' }]
        },
        _embedded: {
          venues: [
            {
              name: 'Pepsi Center',
              type: 'venue',
              id: 'KovZpZAFaJeA',
              test: false,
              url:
                'https://www.ticketmaster.com/pepsi-center-tickets-denver/venue/246112',
              locale: 'en-us',
              images: [
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/v/c8c/314a72bc-c758-42f2-81b3-ca6ea0fc1c8c_414341_SOURCE.jpg',
                  width: 1948,
                  height: 1082,
                  fallback: false,
                  attribution: 'CC BY-SA 2.5 via Wikipedia'
                }
              ],
              distance: 1.57,
              units: 'MILES',
              postalCode: '80204',
              timezone: 'America/Denver',
              city: { name: 'Denver' },
              state: { name: 'Colorado', stateCode: 'CO' },
              country: { name: 'United States Of America', countryCode: 'US' },
              address: { line1: '1000 Chopper Circle' },
              location: { longitude: '-105.010166', latitude: '39.74724' },
              markets: [{ id: '6' }, { id: '51' }],
              dmas: [
                { id: 200 },
                { id: 248 },
                { id: 254 },
                { id: 264 },
                { id: 285 },
                { id: 347 }
              ],
              boxOfficeInfo: {
                phoneNumberDetail:
                  "(303) 405-1111 Information Line Only! The general Pepsi Center information number is (303) 405-8555. For more information you can also go to Pepsi Center's website, www.PepsiCenter.com",
                openHoursDetail:
                  'Winter Hours: Mon - Fri 10AM - 6PM Saturday 10AM - 3PM Sunday Closed If there is an event scheduled on Saturday, Box Office will be open through show time. If there is an event scheduled on Sunday, Box Office will be open.',
                acceptedPaymentDetail:
                  'Cash, Mc, Visa, Discover, AMX. Checks accepted at Avalanche and Nuggets only! No checks for concerts and other events!',
                willCallDetail:
                  'WILL CALL IS AVAILABLE AT THE BOX OFFICE ON THE DAY OF THE SHOW ONLY, STARTING 1 1/2 HOURS BEFORE SHOWTIME.'
              },
              parkingDetail:
                'Parking at Pepsi Center may be easily accessed from I-25 and Auraria Parkway when traveling from the south, or I-25 and Speer Blvd when traveling from the North. Lots start at only $10 per car.',
              accessibleSeatingDetail:
                'Pepsi Center is fully accessible to patrons with disablitities. Ramps are provided at all entries, and elevators will take the patron directly to the concourse level for seating. PATRONS WITH ACCESSIBILITY NEEDS MAY PURCHASE TICKETS FOR AS MANY COMPANIONS AS THEY WISH, UP TO THE TICKET LIMIT ON EACH GIVEN PERFORMANCE.',
              generalInfo: {
                generalRule:
                  'NO: CANS, BOTTLES, CAMERAS, RECORDING EQUIPMENT, FOOD, DRINK, PETS AND WEAPONS.',
                childRule:
                  'Vary by event. Under 2 free - Child must sit on an adults lap.'
              },
              upcomingEvents: { _total: 116, tmr: 105, ticketmaster: 11 },
              _links: {
                self: { href: '/discovery/v2/venues/KovZpZAFaJeA?locale=en-us' }
              }
            }
          ],
          attractions: [
            {
              name: 'P!NK',
              type: 'attraction',
              id: 'K8vZ9171Jo7',
              test: false,
              url: 'https://www.ticketmaster.com/pnk-tickets/artist/718655',
              locale: 'en-us',
              externalLinks: {
                youtube: [
                  { url: 'https://www.youtube.com/user/PinkVideoVault' },
                  { url: 'https://www.youtube.com/user/PinkVEVO' }
                ],
                twitter: [{ url: 'https://twitter.com/Pink' }],
                lastfm: [{ url: 'http://www.last.fm/music/P!nk' }],
                wiki: [{ url: 'https://en.wikipedia.org/wiki/Pink_(singer)' }],
                facebook: [{ url: 'https://www.facebook.com/pink' }],
                musicbrainz: [{ id: 'f4d5cc07-3bc9-4836-9b15-88a08359bc63' }],
                homepage: [{ url: 'http://www.pinkspage.com/' }]
              },
              images: [
                {
                  ratio: '3_2',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg',
                  width: 640,
                  height: 427,
                  fallback: false
                },
                {
                  ratio: '4_3',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_CUSTOM.jpg',
                  width: 305,
                  height: 225,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_16_9.jpg',
                  width: 640,
                  height: 360,
                  fallback: false
                },
                {
                  ratio: '3_2',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_ARTIST_PAGE_3_2.jpg',
                  width: 305,
                  height: 203,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_LANDSCAPE_16_9.jpg',
                  width: 1136,
                  height: 639,
                  fallback: false
                },
                {
                  ratio: '3_2',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_3_2.jpg',
                  width: 1024,
                  height: 683,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RECOMENDATION_16_9.jpg',
                  width: 100,
                  height: 56,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_LARGE_16_9.jpg',
                  width: 2048,
                  height: 1152,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_EVENT_DETAIL_PAGE_16_9.jpg',
                  width: 205,
                  height: 115,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_16_9.jpg',
                  width: 1024,
                  height: 576,
                  fallback: false
                }
              ],
              classifications: [
                {
                  primary: true,
                  segment: { id: 'KZFzniwnSyZfZ7v7nJ', name: 'Music' },
                  genre: { id: 'KnvZfZ7vAeA', name: 'Rock' },
                  subGenre: { id: 'KZazBEonSMnZfZ7v6F1', name: 'Pop' },
                  type: { id: 'KZAyXgnZfZ7v7la', name: 'Individual' },
                  subType: { id: 'KZFzBErXgnZfZ7vAd7', name: 'Musician' },
                  family: false
                }
              ],
              upcomingEvents: {
                _total: 65,
                'mfx-nl': 7,
                tmr: 6,
                ticketmaster: 50,
                'mfx-be': 1,
                'mfx-pl': 1
              },
              _links: {
                self: {
                  href: '/discovery/v2/attractions/K8vZ9171Jo7?locale=en-us'
                }
              }
            }
          ]
        }
      },
      {
        name: 'P!nk',
        type: 'event',
        id: 'Z7r9jZ1AeuM4U',
        test: false,
        url:
          'http://www.ticketsnow.com/InventoryBrowse/TicketList.aspx?PID=2377576',
        locale: 'en-us',
        images: [
          {
            ratio: '3_2',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg',
            width: 640,
            height: 427,
            fallback: false
          },
          {
            ratio: '4_3',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_CUSTOM.jpg',
            width: 305,
            height: 225,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_16_9.jpg',
            width: 640,
            height: 360,
            fallback: false
          },
          {
            ratio: '3_2',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_ARTIST_PAGE_3_2.jpg',
            width: 305,
            height: 203,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_LANDSCAPE_16_9.jpg',
            width: 1136,
            height: 639,
            fallback: false
          },
          {
            ratio: '3_2',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_3_2.jpg',
            width: 1024,
            height: 683,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RECOMENDATION_16_9.jpg',
            width: 100,
            height: 56,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_LARGE_16_9.jpg',
            width: 2048,
            height: 1152,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_EVENT_DETAIL_PAGE_16_9.jpg',
            width: 205,
            height: 115,
            fallback: false
          },
          {
            ratio: '16_9',
            url:
              'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_16_9.jpg',
            width: 1024,
            height: 576,
            fallback: false
          }
        ],
        distance: 1.91,
        units: 'MILES',
        sales: {
          public: {
            startDateTime: '1900-01-01T18:00:00Z',
            startTBD: false,
            endDateTime: '2019-04-02T01:30:00Z'
          }
        },
        dates: {
          start: {
            localDate: '2019-04-01',
            localTime: '19:30:00',
            dateTime: '2019-04-02T01:30:00Z',
            dateTBD: false,
            dateTBA: false,
            timeTBA: false,
            noSpecificTime: false
          },
          status: { code: 'onsale' },
          spanMultipleDays: false
        },
        classifications: [
          {
            primary: true,
            segment: { id: 'KZFzniwnSyZfZ7v7nJ', name: 'Music' },
            genre: { id: 'KnvZfZ7vAev', name: 'Pop' },
            subGenre: { id: 'KZazBEonSMnZfZ7vkEl', name: 'Pop Rock' },
            family: false
          }
        ],
        seatmap: {
          staticUrl:
            'http://resale.ticketmaster.com.au/akamai-content/graphics/TMResale/2/VenueMaps/1315-2670-2-0-PepsiCenterPink.png'
        },
        _links: {
          self: { href: '/discovery/v2/events/Z7r9jZ1AeuM4U?locale=en-us' },
          attractions: [
            { href: '/discovery/v2/attractions/K8vZ9171Jo7?locale=en-us' }
          ],
          venues: [{ href: '/discovery/v2/venues/ZFr9jZe7ek?locale=en-us' }]
        },
        _embedded: {
          venues: [
            {
              name: 'Pepsi Center',
              type: 'venue',
              id: 'ZFr9jZe7ek',
              test: false,
              locale: 'en-us',
              distance: 1.91,
              units: 'MILES',
              postalCode: '80204',
              timezone: 'America/Denver',
              city: { name: 'Denver' },
              state: { name: 'Colorado', stateCode: 'CO' },
              country: { name: 'United States Of America', countryCode: 'US' },
              address: { line1: '1000 Chopper Cir' },
              location: { longitude: '-105.020401', latitude: '39.735001' },
              dmas: [{ id: 264 }],
              upcomingEvents: { _total: 116, tmr: 105, ticketmaster: 11 },
              _links: {
                self: { href: '/discovery/v2/venues/ZFr9jZe7ek?locale=en-us' }
              }
            }
          ],
          attractions: [
            {
              name: 'P!NK',
              type: 'attraction',
              id: 'K8vZ9171Jo7',
              test: false,
              url: 'https://www.ticketmaster.com/pnk-tickets/artist/718655',
              locale: 'en-us',
              externalLinks: {
                youtube: [
                  { url: 'https://www.youtube.com/user/PinkVideoVault' },
                  { url: 'https://www.youtube.com/user/PinkVEVO' }
                ],
                twitter: [{ url: 'https://twitter.com/Pink' }],
                lastfm: [{ url: 'http://www.last.fm/music/P!nk' }],
                wiki: [{ url: 'https://en.wikipedia.org/wiki/Pink_(singer)' }],
                facebook: [{ url: 'https://www.facebook.com/pink' }],
                musicbrainz: [{ id: 'f4d5cc07-3bc9-4836-9b15-88a08359bc63' }],
                homepage: [{ url: 'http://www.pinkspage.com/' }]
              },
              images: [
                {
                  ratio: '3_2',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg',
                  width: 640,
                  height: 427,
                  fallback: false
                },
                {
                  ratio: '4_3',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_CUSTOM.jpg',
                  width: 305,
                  height: 225,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_16_9.jpg',
                  width: 640,
                  height: 360,
                  fallback: false
                },
                {
                  ratio: '3_2',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_ARTIST_PAGE_3_2.jpg',
                  width: 305,
                  height: 203,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_LANDSCAPE_16_9.jpg',
                  width: 1136,
                  height: 639,
                  fallback: false
                },
                {
                  ratio: '3_2',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_3_2.jpg',
                  width: 1024,
                  height: 683,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RECOMENDATION_16_9.jpg',
                  width: 100,
                  height: 56,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_LARGE_16_9.jpg',
                  width: 2048,
                  height: 1152,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_EVENT_DETAIL_PAGE_16_9.jpg',
                  width: 205,
                  height: 115,
                  fallback: false
                },
                {
                  ratio: '16_9',
                  url:
                    'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_TABLET_LANDSCAPE_16_9.jpg',
                  width: 1024,
                  height: 576,
                  fallback: false
                }
              ],
              classifications: [
                {
                  primary: true,
                  segment: { id: 'KZFzniwnSyZfZ7v7nJ', name: 'Music' },
                  genre: { id: 'KnvZfZ7vAeA', name: 'Rock' },
                  subGenre: { id: 'KZazBEonSMnZfZ7v6F1', name: 'Pop' },
                  type: { id: 'KZAyXgnZfZ7v7la', name: 'Individual' },
                  subType: { id: 'KZFzBErXgnZfZ7vAd7', name: 'Musician' },
                  family: false
                }
              ],
              upcomingEvents: {
                _total: 65,
                'mfx-nl': 7,
                tmr: 6,
                ticketmaster: 50,
                'mfx-be': 1,
                'mfx-pl': 1
              },
              _links: {
                self: {
                  href: '/discovery/v2/attractions/K8vZ9171Jo7?locale=en-us'
                }
              }
            }
          ]
        }
      }
    ]
  },
  _links: {
    first: {
      href:
        '/discovery/v2/events.json?radius=200&geoPoint=9xj6518xn&page=0&size=2'
    },
    self: {
      href: '/discovery/v2/events.json?size=2&radius=200&geoPoint=9xj6518xn'
    },
    next: {
      href:
        '/discovery/v2/events.json?radius=200&geoPoint=9xj6518xn&page=1&size=2'
    },
    last: {
      href:
        '/discovery/v2/events.json?radius=200&geoPoint=9xj6518xn&page=495&size=2'
    }
  },
  page: { size: 2, totalElements: 991, totalPages: 496, number: 0 }
};

export const userWatchlist = [
  {
    id: 3,
    name: 'EnParejaDos - con Adrian Uribe y Consuelo Duval',
    e_id: 'G5vzZ48H7Q7cH',
    url: 'https://www.altitudetickets.com/events/detail/adrian-uribe',
    img:
      'https://s1.ticketm.net/dam/a/b1e/5d94da98-9bde-4608-a1ed-937950668b1e_790071_CUSTOM.jpg',
    date: 'Fri, Nov 30, 2018 8:00 PM',
    venue_name: 'Paramount Theatre',
    address: '1621 Glenarm Denver, CO 80202',
    lat: '39.744362',
    lng: '-104.990271',
    distance: '0.55',
    created_at: '2018-10-26T19:22:39.218Z',
    updated_at: '2018-10-26T19:22:39.218Z'
  }
];
