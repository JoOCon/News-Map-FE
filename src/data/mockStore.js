import * as user from './mockUser';
import * as events from './mockEvents';

export const mockStore = {
  activeUser: user.mockStateUser,
  userLocation: { latitude: 39.43243, longitude: -104.98434 },
  watchList: events.userWatchlist,
  events: events.eventsResponse
};
