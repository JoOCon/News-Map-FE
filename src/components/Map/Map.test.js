import React from 'react';

import { Map } from './Map';
import * as events from '../../data/mockEvents';
import moment from 'moment';

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
  Map: () => ({})
}));

describe('Map', () => {
  let wrapper;
  let mockLat;
  let mockLng;

  beforeEach(() => {
    mockLat = 35.2323;
    mockLng = -105.3434;
    wrapper = shallow(
      <Map
        latitude={mockLat}
        longitude={mockLng}
        events={events.eventsResponse}
        setEvents={jest.fn()}
      />
    );
  });

  it('should matchSnapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call window.fetch with correct Params', () => {
    navigator.geolocation = {
      getCurrentPosition: () => {
        return { location: { coords: { latitude: 23, longitude: -104 } } };
      }
    };

    wrapper.instance().setLatLngEvents();

    expect(wrapper.state().latitude).toEqual(0);
    expect(wrapper.state().longitude).toEqual(0);
  });

  it('should retrieve events', async () => {
    window.fetch = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        json: () => Promise.resolve(events.mockEventsResponse)
      });
    });

    wrapper.instance().retrieveEvents(35, -103);

    expect(window.fetch).toHaveBeenCalled();
  });
});
