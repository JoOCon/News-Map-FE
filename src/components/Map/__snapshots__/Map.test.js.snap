// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map should matchSnapshot 1`] = `undefined`;
